src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -186,14 +188,29 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
+                        {
+                            length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                            if (length == chars.Length)
+                            {
+                                char[] toReturn = chars;
+                                chars = null;
+                                ArrayPool<char>.Shared.Return(toReturn);
```

Comment:
what is the purpose of `toReturn` - can't you do:
```c#
                               ArrayPool<char>.Shared.Return(chars);
                                minimumLength = Math.Min(minimumLength * 2, short.MaxValue);
                                chars = ArrayPool<char>.Shared.Rent(minimumLength);
```

Reply:
@danmosemsft This would call `Return` multiple times` when `Rent` throws an exception.

It would be fine to write it this way if the `Return` is not inside finally block (ie we let GC take care of collecting the buffer when exception is thrown).

Reply:
ah yes - tricky. might be worth a comment.